src/modules/navigator/rtl.cpp

```diff
@@ -228,8 +228,9 @@ void RTL::find_RTL_destination()
 		}
 	}
 
-	if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
-		_rtl_alt = calculate_return_alt_from_cone_half_angle((float)_param_rtl_cone_half_angle_deg.get());
+	// For a fixed wing, if we are close to target and inside the cone shape, limit the RTL altitude to current altitude (due to vehicle dynamics)
+	if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+		_rtl_alt = calculate_return_alt_from_cone_half_angle(_param_rtl_cone_half_angle_deg.get());
```

Comment:
I think this has undesired behavior for FW to use the cone. Imagine being close over ground when RTL gets triggered, do you then want it to loiter around in a very low altitude? The cone for MC only works because you can guarantee that the drone won't leave the cone for the RTL, which you certainly can't guarantee.